packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/quanticCategoryFacet.html

```diff
@@ -11,84 +14,139 @@
             <div onchange={handleKeyUp} class="slds-size_1-of-1">
-              <lightning-input class="facet__searchbox-input slds-p-horizontal_x-small slds-var-p-bottom_x-small" name={labels.search} type="search" label="facet-input" variant="label-hidden" placeholder={labels.search}></lightning-input>
+              <lightning-input
+                class="facet__searchbox-input slds-p-horizontal_x-small slds-var-p-bottom_x-small"
+                name={labels.search}
+                type="search"
+                label="facet-input"
+                variant="label-hidden"
+                placeholder={labels.search}
+              ></lightning-input>
             </div>
           </template>
+
           <template if:false={isFacetSearchActive}>
-            <template if:true={hasParents}>
-              <li class="facet__value-option slds-size_1-of-1 slds-p-around_x-small">
-                <div class="facet__allCategories slds-grid slds-grid_vertical-align-center" onclick={reset}>
-                  <lightning-icon class="slds-m-right_xxx-small" icon-name="utility:chevronleft" size="xx-small"></lightning-icon>
-                  <span class="nowrap">{labels.allCategories}</span>
-                </div> 
-              </li> 
-            </template>
-            <template for:each={nonActiveParents} for:item="v">
-              <c-quantic-category-facet-value onselectvalue={onSelectValue} key={v.value} item={v} non-active-parent="true"></c-quantic-category-facet-value>
-            </template>
-            <template if:true={activeParent}>
-              <li class="facet__active-parent slds-grid slds-grid_vertical-align-center slds-p-around_x-small slds-m-left_medium">
-                <span class="nowrap parent-text__bold">{activeParent.value}</span>
-                <span class="facet__number-of-results parent-text__bold slds-var-m-left_xx-small">({activeParent.numberOfResults})</span>
-              </li>  
-            </template>
-            <li class="slds-grid">
-              <ul class="slds-size_1-of-1">
+            <fieldset>
+              <legend class="slds-assistive-text">{field}</legend>
+              <ul>
+                <template if:true={hasParents}>
+                  <li
+                    class="facet__value-option slds-size_1-of-1 slds-p-around_x-small"
+                  >
+                    <div
+                      class="facet__allCategories slds-grid slds-grid_vertical-align-center"
+                      onclick={reset}
+                    >
+                      <lightning-icon
+                        class="slds-m-right_xxx-small"
+                        icon-name="utility:chevronleft"
+                        size="xx-small"
+                      ></lightning-icon>
+                      <span class="nowrap">{labels.allCategories}</span>
+                    </div>
+                  </li>
+                </template>
+                <template for:each={nonActiveParents} for:item="v">
```

Comment:
If possible, I would recommend rendering results as a hierarchy, which should improve accessibility. This is what we tried to do in https://github.com/coveo/ui-kit/pull/2029.

Reply:
Good idea but out of scope for this PR.